### PR TITLE
feat: Angular boilerplate

### PR DESCRIPTION
**Pull request by 9kubczas4**
PR includes a basic Angular Hello World boilerplate for Firebase App Hosting.

It shows basic features like:
- SSR
- SSG
- Deferrable Views

### CODE REVIEW

File: starters/angular/basic/src/app/pages/home/home.component.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    <h2>Deferrable Views</h2>
+    <p>
+      Take advantage of one of the most important features of Angular from recent releases.
+      Deferrable views are a powerful tool that can be used to reduce the initial bundle size of your application or defer heavy components that may not ever be loaded until a later time.
+      Visit <a routerLink="deferrable-views">/defer</a> page to see defer in Action.
+    </p>
+  </article>
+</section>
+
+<section class="links-container">
```
Comment (9kubczas4) [inline starters/angular/basic/src/app/pages/home/home.component.html:25]: @jhuleatt @jamesdaniels please let me know what links should be displayed here.

Comment (leoortizz) [reply starters/angular/basic/src/app/pages/home/home.component.html:25]: FWIW the links were removed in the Next.js boilerplate

Comment (jhuleatt) [reply starters/angular/basic/src/app/pages/home/home.component.html:25]: let's remove the links, but keep the CSS please so we can easily add them back in later.

**Review by jhuleatt — APPROVED** [review]
Looks good to me once links are removed (https://github.com/FirebaseExtended/firebase-framework-tools/pull/174#discussion_r1591081364). 

Seems like CI is failing, is that an easy fix?

**Comment by Yuangwang** [comment]
Could you add a brief description of what this is for?

**Comment by 9kubczas4** [comment]
@Yuangwang just added a short description to the PR.

**Review by Yuangwang — APPROVED** [review]
Two small comments, but otherwise LGTM once the comments are addressed and CI is fixed

File: starters/angular/basic/src/app/pages/home/home.component.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<h1 class="large-heading">Hello, My App</h1>
```
[the remark this answers is not in the document]
Comment (9kubczas4) [reply starters/angular/basic/src/app/pages/home/home.component.html:1]: Hi,

It was implemented based on designs.
I'm more than happy to introduce some content changes, but I just wanted to be sure that all of us are on the same page.

CC: @jhuleatt

[the remark this answers is not in the document]
Comment (leoortizz) [reply starters/angular/basic/src/app/pages/home/home.component.html:1]: +1 to mirror the Next.js h1
https://github.com/FirebaseExtended/firebase-framework-tools/blob/e821affa73cf8485482cafd324e76831abe56104/starters/nextjs/basic/src/app/page.tsx#L6

File: starters/angular/basic/src/app/pages/ssg/ssg.component.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<h1 class="heading">Sample title for SSG</h1>
```
[the remark this answers is not in the document]
Comment (leoortizz) [reply starters/angular/basic/src/app/pages/ssg/ssg.component.html:1]: I changed the Next.js one to just `SSG` so I think this one can follow
https://github.com/FirebaseExtended/firebase-framework-tools/blob/e821affa73cf8485482cafd324e76831abe56104/starters/nextjs/basic/src/app/ssg/page.tsx#L6

File: starters/angular/basic/src/styles.scss [hunk 4]
```diff
@@ -244,39 +244,12 @@ a:hover {
   color: var(--secondary-contrast);
   letter-spacing: -1%;
 }
-.link,
```
Comment (leoortizz) [inline starters/angular/basic/src/styles.scss]: @9kubczas4 we decided to keep the links CSS, can you revert? See https://chat.google.com/room/AAAA6WrZdos/QT6YN1rqe3E/7FKOjctpQuM?cls=10

Comment (9kubczas4) [reply starters/angular/basic/src/styles.scss]: Oh, good catch! ;) Will do.

Comment (9kubczas4) [reply starters/angular/basic/src/styles.scss]: Done. @leoortizz

File: starters/angular/basic/src/app/pages/home/home.component.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (9kubczas4) [reply starters/angular/basic/src/app/pages/home/home.component.html]: I've applied proposal in recent commit.

File: starters/angular/basic/src/app/pages/ssg/ssg.component.html (hunk 3 above)
[the remark this answers is not in the document]
Comment (9kubczas4) [reply starters/angular/basic/src/app/pages/ssg/ssg.component.html]: I've applied proposal in recent commit.

**Review by sjjj986 — APPROVED** [review]

File: starters/angular/basic/server.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  const port = process.env['PORT'] || 4000;
+
+  // Start up the Node server
+  const server = app();
+  server.listen(port, () => {
+    console.log(`Node Express server listening on http://localhost:${port}`);
+  });
+}
+
+// Note: The express server is started by firebase automatically.
```
Comment (leoortizz) [inline starters/angular/basic/server.ts:56]: nit
```suggestion
// Note: The express server is started by Firebase automatically.
```

**Comment by 9kubczas4** [comment]
Hi @Yuangwang,

The build step is failing on CI, locally it works fine.

The following error occurs:

```
Error: src/firebase-aware.ts(8,17): error TS7016: Could not find a declaration file for module 'lru-cache'. '/home/runner/work/firebase-framework-tools/firebase-framework-tools/node_modules/lru-cache/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/lru-cache` if it exists or add a new declaration (.d.ts) file containing `declare module 'lru-cache';`
Error: src/firebase-aware.ts(23,13): error TS7006: Parameter 'value' implicitly has an 'any' type.
npm ERR! Lifecycle script `build` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: firebase-frameworks@0.11.2 
npm ERR!   at location: /home/runner/work/firebase-framework-tools/firebase-framework-tools/packages/firebase-frameworks 
```

It looks like it's unrelated to changes in this PR.

File: starters/angular/basic/README.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# FirebaseAppHostingAngular
```
Comment (leoortizz) [inline starters/angular/basic/README.md:1]:
```suggestion
# Angular on Firebase App Hosting
```

**Review by leoortizz — APPROVED** [review]

**Review by rufustent-dotcom — APPROVED** [review]

**Pull request merged**